chain/stmgr/forks.go

```diff
@@ -208,7 +211,21 @@ func (sm *StateManager) handleStateForks(ctx context.Context, root cid.Cid, heig
 	return retCid, nil
 }
 
-func (sm *StateManager) hasExpensiveFork(ctx context.Context, height abi.ChainEpoch) bool {
+// Returns true if executing the current tipset would trigger an expensive fork.
+//
+// - If the tipset is the genesis, this function always returns false.
+// - If inclusive is true, this function will also return true if applying a message on-top-of the
+//   tipset would trigger a fork.
+func (sm *StateManager) hasExpensiveForkBetween(parent, height abi.ChainEpoch) bool {
+	for h := parent; h < height; h++ {
+		if _, ok := sm.expensiveUpgrades[h]; ok {
```

Comment:
This probably doesn't show up unless you force it but I believe the number of epochs between parent and height can be made arbitrarily big from the cli and it would be nice for this not to hang by taking advantage of sparseness of upgrade heights in the `expensiveUprages` map.

Reply:
Oh ok I see now that the only uses of this are constrained so that the number of iterations is bounded by number of null blocks.  There's potential for misuse from future code maybe (though parameter names help) but no hanging possible.

Reply:
I can have a switch if the number of epochs exceeds the number of upgrades.

Reply:
Hm. Actually, I'd rather not. It introduces an additional code path that won't be well tested.

Reply:
(I could write tests, but I'd rather stick with one code path)